src/components/z-carousel/index.tsx

```diff
@@ -12,25 +13,129 @@ import { Component, h, Prop } from "@stencil/core";
+    console.debug('z-carousel: indexChange', this.itemIndex);
+  }
+
+  @Watch('itemIndex')
+  onIndexChange() {
+    this.items.find((a) => !a.classList.contains('hidden'))?.classList.add('hidden');
+    this.items[this.itemIndex].classList.remove('hidden');
+    this.items[this.itemIndex].scrollIntoView({
+      behavior: 'smooth',
+    });
+    this.emitIndexChange();
+  }
+
+  componentDidLoad() {
+    this.itemsContainer = this.hostElement.querySelector('.z-carousel-items-container');
+    if (!this.itemsContainer) {
+      return;
+    }
+    this.items = Array.from(this.itemsContainer.querySelectorAll('li'));
+  }
+
+  onPrev() {
+    if (this.single) {
+      this.itemIndex = Math.max(0, this.itemIndex - 1);
+      return;
+    }
+
+    const scroller = this.itemsContainer;
+    if (!scroller) {
+      return;
+    }
+
+    scroller.scrollBy({
+      left: -scroller.clientWidth / 2,
+      behavior: 'smooth',
+    });
+  }
+
+  onNext() {
+    if (this.single) {
+      this.itemIndex = Math.min(this.itemIndex + 1, this.items.length - 1);
+      return;
+    }
+
+    const scroller = this.itemsContainer;
+    if (!scroller) {
+      return;
+    }
+
+    scroller.scrollBy({
+      left: scroller.clientWidth / 2,
+      behavior: 'smooth',
+    });
+  }
+
+  showFooter() {
+    return this.buttons === CarouselButtonOptions.BOTTOM ||
+      this.progress === CarouselProgressOptions.DOTS || this.progress === CarouselProgressOptions.NUMBERS;
+  }
+
   render() {
     if (this.isLoading) {
       return (
         <div style={{ height: `${this.ghostLoadingHeight}px` }}>
           <z-ghost-loading></z-ghost-loading>
-          <div style={{ display: "none" }}>
+          <div class="loading-items-container">
             <slot />
           </div>
         </div>
       );
     }
-    return (
-      <ul class="z-carousel-items-container">
-        <slot />
-      </ul>
-    );
+
+    return [
+      <div class="z-carousel-container">
+        {this.buttons === CarouselButtonOptions.TOP && <z-button data-direction='prev' icon='chevron-left' onClick={this.onPrev.bind(this)}/>}
+        <ul class="z-carousel-items-container">
+          <slot />
+        </ul>
+        {this.buttons === CarouselButtonOptions.TOP && <z-button data-direction='next' icon='chevron-right' onClick={this.onNext.bind(this)}/>}
+      </div>,
+      this.showFooter() && <div class="z-carousel-footer">
+        {this.buttons === CarouselButtonOptions.BOTTOM && <z-button variant={ButtonVariantEnum.tertiary} icon='arrow-left-filled' onClick={this.onPrev.bind(this)}/>}
+        {this.progress === CarouselProgressOptions.DOTS && this.single && this.items && <div class="dots-progress">
+          {this.items.map((_item, key) => this.itemIndex === key ? <button class="current"><z-icon name="black-circle-filled"/></button> :
+            <button onClick={() => this.itemIndex = key}><z-icon name="white-circle-filled"/></button>)}
```

Comment:
le z-icon dei dots mi sa che vanno in gray800 (o qualche token di tema che di default è quel gray)

Reply:
sistemati gli stili del dots indicator in generale, ora è uguale al PG